### DISCUSSION TITLE
How to use Sample format / Sample rate converter (Doubt)

### DISCUSSION THREAD

**Discussion by Cultura-Difusa** [enzo1982/freac#370] [Help & Support] [answered]
Hello there!

Firstly let me thank the Enzo for his work. Thank you.

Now my question:

> Do the signal processing plugins (Sample format and Sample converter) get applied when enabled to all conversions indistinctly, or only to input files with different sample rate/format to the desired output.

That is it.

Thanks in advance!

**Comment by enzo1982** [comment]
fre:ac always changes the sample format to 32 bit float when the sample rate converter is active, even if the source sample rate already matches the target. I have plans to change that in a future update, though.

The sample format converter on the other hand does not change anything when the source format already matches the target format.

**Reply by Cultura-Difusa** [reply]
Sorry to bother you.

You mention:

___

> [...] always changes the sample format to 32 bit float when the sample rate converter is active, [...]

and

>  [...] sample format converter on the other hand does not change anything when the source format already matches the [...]

____

____

1. **Correct me if I am wrong;** but do you mean that?:

> _If **sampling rate** of the source is **different** from  the output (with sampling rate plugin enabled), **then**, gets converted. And that **being all things equal, it does not.**-_

and (that)

> _Regardless of the input **sample format**, if the plugin is enabled (sample format), the **output** is **allways** changed_.

___

2. **And yet, if the previous is correct, I still have a question:**

Does not the "sample resolution dial" (8-32) affect the output?

I am asking these questions, because I have some lossless files from **people who think more than** 44 khz and 16bit are "necessary", and **I want to resample those files to sane rates/formats** (16/44) doing the less harm possible.

Thanks in advance!

**Reply by Lee-Carre** [reply, marked as answer]
@Cultura-Difusa

> I have some lossless files from **people who think more than** 44 khz and 16bit are “necessary”, and **I want to resample those files to sane rates/formats** (16/44) doing the le[ast] harm possible.

Now that you've stated your actual goal, and relevant context, that helps a lot.

You want the `sample rate converter` filter, **not** the `sample format converter`. They do different things.

> 1. **Correct me if I am wrong;** but do you mean that?:
> 
> > _If **sampling rate** of the source is **different** from  the output (with sampling rate plugin enabled), **then**, gets converted. And that **being all things equal, it does not.**-_
> 
> […]

Based on my reading of Enzo1982's reply, above, this is incorrect. The sample rate converter **always** converts, even when source & target already match. Converting to 32-bit float (and back) will likely result in rounding errors, and the converted file won't match the original file.

So, instead, you'll want to determine which files have sample rates/resolutions which are greater than you'd like (perhaps use `FFprobe`, part of `FFmpeg`), and only have the sample rate converter filter enabled for those files (the ones from friends). The rest of the time, you want both the sample filters disabled.

Might I suggest setting up different configuration profiles; one for non-conversion, and one for conversion, so that switching between them is easy (and it makes it more obvious if you've got the converter active, in case you don't want it enabled).

> […]
> 
> > _Regardless of the input **sample format**, if the plugin is enabled (sample format), the **output** is **allways** changed_.

Again, based on my reading of Enzo1982's reply, this is incorrect, for the sample **format** converter (but would be true/correct if you'd asked about the sample **rate** converter).

So, if you want to leave 44kHz@16-bit files untouched (except for whatever transcoding you're doing), then you want to ensure that the sample **rate** converter is disabled. Since what you're trying to do requires a rate conversion, not a format conversion, you want the sample format converter disabled at all times.

> Does not the "sample resolution dial" (8-32) affect the output?

Yes, it would. This would be the bits-per-sample value.

The exact effect it has, depends on if you're changing the **rate** or merely the **format**. Again, different things.

> Thanks in advance!

This presumes that you're gonna get an answer, or any response.

---

> […] **people who think more than** 44 khz and 16bit are “necessary” […]

Direct them to [watch Episode #2 (Digital Show and Tell) of Monty's Xiph tutorial videos](//www.xiph.org/video/), in which he thoroughly debunks (with demonstrations, using analogue test equipment) this fallacy, and explains what >48kHz and >24-bit(-per-sample) are actually for.

Also point out that, unless they're encoding losslessly from a high-resolution source (directly from an ADC outputting >44kHz and/or >16-bit, else transcoding from other files which were generated that way), then they're just ‘up-converting’ which makes the files larger without actually adding any fidelity (same as interpolating an image to increase the pixel-count; software can't add detail which isn't already in the original/source file). If so (up-converting), they're simply wasting resources (CPU time, storage, network capacity) with no gain.

Your optimal solution, is to convince/persuade these friends of their folly, so that they don't generate bloated files in the first place. Then, no need for any conversion, and everyone benefits (including the friends) 😀.

**Reply by Cultura-Difusa** [reply]
Thank you, I will re-read your answers, understand properly what I want, and set up my rig/profiles accordingly.

Again, thanks.
